### PR TITLE
#3812 Paper Topic and Stream action with Topic api and Stream api (cr…

### PR DESCRIPTION
**Pull request by wu87988622**
…eate, start, stop, remove, update)

### Checklist
- [x] [Review Contribution Guidelines](https://ohara.readthedocs.io/en/latest/contributing.html)
- [x] Add Reviewers (see [Authors](https://github.com/oharastream/ohara#ohara-team))
- [x] Add Assignees (of course, you are the assignee by default!)
- [x] DON'T set a Milestone
- [x] DON'T add any labels (PR labels will be added automatically by Jenkins)
- [x] Link to an issue(You can do so by using the required by keyword: required by #3812 )
- [ ] Add unit tests (or please explain why this PR is not worth having new tests)
- [ ] Pass QA (Sometimes you may encounter an existing flaky test which would obstruct you from getting QA passed. If the failed tests are unrelated, you can add a comment in the PR thread)

### CODE REVIEW

File: ohara-manager/client/src/components/Pipeline/Pipeline.js [hunk 1]
```diff
@@ -167,15 +212,179 @@ const Pipeline = () => {
...
+                      topic,
+                      firstLink,
+                      secondeLink,
+                    } = pipelineUtils.utils.getConnectionOrder(cells);
+                    let topicRes;
+                    let sourceRes;
+                    let sinkRes;
+                    let streamRes;
+                    switch (type) {
+                      case CONNECTION_ORDER.source_topic:
```
Comment (eechih) [inline ohara-manager/client/src/components/Pipeline/Pipeline.js]: **type** 與 **CONNECTION_ORDER** 有點衝突~~ 改成 CONNECTION_TYPE 會不會比較好？

Comment (wu87988622) [reply ohara-manager/client/src/components/Pipeline/Pipeline.js]: nice

File: ohara-manager/client/src/components/Pipeline/Paper/PaperUtils.js [hunk 2]
```diff
@@ -223,6 +223,7 @@ export const createConnection = params => {
           x: topicX,
           y: topicY,
         },
+        shouldSkipOnElementAdd: true,
```
Comment (konekoya) [inline ohara-manager/client/src/components/Pipeline/Paper/PaperUtils.js]: Thanks 💯

File: ohara-manager/client/src/components/Pipeline/Pipeline.js [hunk 3]
```diff
@@ -68,14 +68,53 @@ const Pipeline = () => {
...
+  const {
+    create: createStream,
+    updateLinkTo: updateStreamLinkTo,
+    updateLinkFrom: updateStreamLinkFrom,
+    start: startStream,
+    stop: stopStream,
+    remove: removeStream,
+    removeLinkTo: removeStreamLinkTo,
+    removeLinkFrom: removeStreamLinkFrom,
+  } = pipelineUtils.stream();
```
Comment (konekoya) [inline ohara-manager/client/src/components/Pipeline/Pipeline.js]: You can also do `const stremUtils = pipelineUtils.stream();`

Then use it like `stremUtils.create()`. This avoids the naming conflicts so you're still able to use the original method name thru out the file but also saves your time of renaming each method one by one :)

File: ohara-manager/client/src/components/Pipeline/PipelineApiHelper/apiHelperUtils.js [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+
+export const getCellJarKey = cell => {
+  return get(cell, 'attributes.jarKey', null);
+};
+
+export const getCellState = cell => {
+  return get(cell, 'data.state', CELL_STATUS.stopped);
+};
+
+export const getConnectionOrder = cells => {
```
Comment (konekoya) [inline ohara-manager/client/src/components/Pipeline/PipelineApiHelper/apiHelperUtils.js]: 😱 super duper

File: ohara-manager/client/src/components/Pipeline/Toolbox/ToolboxHooks.js [hunk 5]
```diff
@@ -60,7 +59,9 @@ export const useFiles = () => {
           name: file.name,
           group: file.group,
           classInfos: file.classInfos.filter(
-            classInfo => classInfo.kind === KIND.stream,
+            classInfo =>
+              classInfo.settingDefinitions.find(def => def.key === 'kind')
+                .defaultValue === KIND.stream,
```
Comment (konekoya) [inline ohara-manager/client/src/components/Pipeline/Toolbox/ToolboxHooks.js]: ❤️

File: ohara-manager/client/src/context/stream/streamActions.js [hunk 6]
```diff
@@ -30,6 +30,7 @@ export const createActions = context => {
         return action.success(data);
       } catch (e) {
         dispatch(routine.failure(e.message));
+        return action.failure(e.message);
```
Comment (konekoya) [inline ohara-manager/client/src/context/stream/streamActions.js]: QQ

**Pull request merged**